lib/chef/application/client.rb

```diff
@@ -384,6 +338,125 @@ def run_application
   end
 
   private
+
+  def configure_local_mode
+    update_local_mode
+    update_chef_repo_path
+    fetch_local_mode_recipes!
+  end
+
+  def verify_audit_mode!
+    mode = config[:audit_mode] || Chef::Config[:audit_mode]
+
+    return unless mode
+
+    assert_valid_audit_mode!(mode)
+    # This should be removed when audit-mode is enabled by default/no longer
+    # an experimental feature.
+    Chef::Log.warn(audit_mode_experimental_message) unless mode == :disabled
+  end
+
+  def assert_valid_audit_mode!(mode)
+    error_message = unrecognized_audit_mode(mode)
+    valid_mode = %i(enabled disabled audit_only).include?(mode)
+
+    Chef::Application.fatal!(error_message) unless valid_mode
```

Comment:
This almost introduces more cognitive overhead than the one-liner.  The error_message variable doesn't really help me read it.